File: app/src/main/java/com/ed/redditapp/ui/subreddit/SubredditPresenter.java
package com.ed.redditapp.ui.subreddit;

import com.ed.redditapp.ui.postlist.Post;

public class SubredditPresenter {
    private final SubredditActivity view;
    private final SubredditModel model;

    public SubredditPresenter(SubredditActivity view, SubredditModel model) {
        this.view = view;
        this.model = model;
    }

    public void onActivityCreated(String subredditName) {
        model.getSubredditInfo(subredditName, view::updateSubredditInfo);
        model.getSubredditPosts(subredditName, view::updatePosts);
    }

    public void onPostTitleClicked(Post post) {
        view.gotoPostDetail(post);
    }
}

File: app/src/main/java/com/ed/redditapp/ui/subreddit/SubredditPresenter.kt
package com.ed.redditapp.ui.subreddit

import com.ed.redditapp.lib.api.Post
import com.ed.redditapp.lib.api.PostContentType

class SubredditPresenter(
        private val view: SubredditActivity,
        private val model: SubredditModel) {

    fun onActivityCreated(subredditName: String) {
        model.getSubredditInfo(subredditName, view::updateSubredditInfo)
        model.getSubredditPosts(subredditName, view::updatePosts)
    }

    fun onPostTitleClicked(post: Post) {
        view.gotoPostDetail(post)
    }

    fun onThumbnailClicked(post: Post) {
        if (post.contentType == PostContentType.IMAGE) {
            view.gotoImageView(post.thumbnailSource!!.url)
        }

        if (post.contentType == PostContentType.VIDEO_HOSTED) {
            view.gotoVideoView(post.content)
        }
    }
}